init-tools.sh

```diff
@@ -15,6 +15,43 @@ __PROJECT_JSON_FILE=$__PROJECT_JSON_PATH/project.json
 __PROJECT_JSON_CONTENTS="{ \"dependencies\": { \"Microsoft.DotNet.BuildTools\": \"$__BUILD_TOOLS_PACKAGE_VERSION\" }, \"frameworks\": { \"netcoreapp1.0\": { } } }"
 __INIT_TOOLS_DONE_MARKER=$__PROJECT_JSON_PATH/done
 
+# Modified platform detection logic from dotnet/cli/scripts/obtain/dotnet-install.sh 16692fc
```

Comment:
Since the implementation of Run, the platform detection code is done by the tool and not the scripts. We've been reducing the code from the scripts which is good, however we can't use the same approach here.

@dagood If we are going to need this platform detection in `init-tools` could we do it in the `init-tools` that comes from Build Tools and pass the value to the repo `init-tools`? That way all the repos share the same code